brambling/api/v1/views.py

```diff
@@ -155,6 +157,37 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data, status=status.HTTP_201_CREATED if created else status.HTTP_200_OK)
 
 
+class OrderSearchViewSet(viewsets.ReadOnlyModelViewSet):
+    "A ViewSet that filters orders based on a single search term."
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    filter_backends = (filters.SearchFilter,)
+    permission_classes = [OrderSearchPermission]
+    search_fields = ("code", "person__given_name", "person__middle_name",
+        "person__surname", "attendees__given_name",
+        "attendees__middle_name", "attendees__surname")
+
+    def get_event(self):
+        event_id = self.request.query_params.get('event', None)
+        return Event.objects.get(pk=event_id)
```

Comment:
This will throw an error if event_id is None. Better to wrap in an if and return None otherwise. Or `return None if event_id is None else Event.etc` if you want a one-liner.